src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1476,7 +1476,8 @@ private static bool RequiresVariableReceiver(BoundExpression receiver, Symbol sy
         {
             return symbol.RequiresInstanceReceiver()
                 && symbol.Kind != SymbolKind.Event
-                && receiver?.Type?.IsValueType == true;
+                && receiver?.Type?.IsValueType == true
+                && !symbol.ContainingType.IsExtension;
```

Comment:
>&& !symbol.ContainingType.IsExtension; [](http://example.com/codeflow?start=16&length=38)

It would be good to add a comment explaining why this is important.

Reply:
BTW, does this need a spec update?

Reply:
Filed [https://github.com/dotnet/csharpstandard/issues/1078](https://github.com/dotnet/csharpstandard/issues/1078) for spec bug: 
1. should check receiver instead of declaring type of property, 
2. should likely check for "not a reference type" instead of "a struct type".

Reply:
Filed [https://github.com/dotnet/roslyn/issues/73003](https://github.com/dotnet/roslyn/issues/73003) for likely implementation bug (should check for "not a reference type" instead)

Also added a note to extensions spec (the error is applicable even when the property is declared in an extension type): [https://github.com/dotnet/csharplang/pull/8012/commits/47f6ad679a349d394c0bb9ff5547c8a938280049](https://github.com/dotnet/csharplang/pull/8012/commits/47f6ad679a349d394c0bb9ff5547c8a938280049)